modules/fluid_properties/src/userobjects/BrineFluidProperties.C

```diff
@@ -59,7 +60,8 @@ BrineFluidProperties::BrineFluidProperties(const InputParameters & parameters)
   {
     const std::string class_name = "NaClFluidProperties";
     InputParameters params = _app.getFactory().getValidParams(class_name);
-    _fe_problem.addUserObject(class_name, nacl_name, params);
+    if (!_fe_problem.hasUserObject(nacl_name)) // multi-threaded guard
+      _fe_problem.addUserObject(class_name, nacl_name, params);
```

Comment:
Same issue here - need to only run this on thread zero or have some sort of shared mutex.